yt/frontends/open_pmd/data_structures.py

```diff
@@ -502,18 +557,85 @@ def _is_valid(self, *args, **kwargs):
         except (IOError, OSError, ImportError):
             return False
 
-        requirements = ["openPMD", "basePath", "meshesPath", "particlesPath"]
         attrs = list(f["/"].attrs.keys())
-        for i in requirements:
+        for i in opmd_required_attributes:
             if i not in attrs:
                 f.close()
                 return False
 
-        known_versions = [StrictVersion("1.0.0"),
-                          StrictVersion("1.0.1")]
-        if StrictVersion(f.attrs["openPMD"].decode()) in known_versions:
+        if StrictVersion(f.attrs["openPMD"].decode()) not in ompd_known_versions:
+            f.close()
+            return False
+
+        if f.attrs["iterationEncoding"].decode() == "fileBased":
             f.close()
             return True
+
+        f.close()
```

Comment:
I *think* this is causing the test failure?

I'd try refactoring the use of file handles here to use a context manager to avoid needing to explicitly close the handle.